### ISSUE TITLE
Onnx MaxPool op does not work with tensorflow max_pool_with_argmax op

### ISSUE THREAD

**Issue by mpaillassa** [onnx/onnx#3356]
# Bug Report

### Describe the bug
I think this issue is in between a bug and a missing feature.
 
I am trying to convert and use for inference a keras model that uses the tensorflow [max_pool_with_argmax](https://www.tensorflow.org/api_docs/python/tf/nn/max_pool_with_argmax) op with keras2onnx.
When converting it prints this warning:
`WARN: No corresponding ONNX op matches the tf.op node detector/StatefulPartitionedCall/StatefulPartitionedCall/max_pool/MaxPoolWithArgmax of type MaxPoolWithArgmax
      The generated ONNX model needs run with the custom op supports.`
And during inference it fails with:
`onnxruntime.capi.onnxruntime_pybind11_state.Fail: [ONNXRuntimeError] : 1 : FAIL : Load model from from_keras.onnx failed:Fatal error: MaxPoolWithArgmax is not a registered function/op`
When looking at the onnx [MaxPool op doc](https://github.com/onnx/onnx/blob/master/docs/Operators.md#MaxPool) it states it can return the pooled indices.
I guess it works for the torch pooling operation but not the tensorflow one (I have not checked for it yet). 
Would it be possible to enable the conversion of the tensorflow max_pool_with_argmax op to the onnx MaxPool op as they perform the same operation ?

### System information
- OS Platform and Distribution:  Ubuntu 20.04
- ONNX version:  1.8.1 (keras2onnx 1.7.0 and onnxruntime 1.7.0)
- Python version: 3.8.5
- GCC/Compiler version (if compiling from source):
- CMake version:
- Protobuf version: 
- Visual Studio version (if applicable):

### Reproduction instructions
- Describe the code to reproduce the behavior.
```
import tensorflow as tf
import numpy as np

class Bug(tf.keras.Model):
    
    def __init__(self):
        super(Bug, self).__init__()

    def call(self, inputs):
        p, i = tf.nn.max_pool_with_argmax(inputs, ksize=[1, 2, 2, 1], strides=[1, 2, 2, 1], padding='SAME')
        return p

b = Bug()
inp = np.random.randint(0, 10, (1, 4, 4, 1))
r = b(inp)
print(inp[0,:,:,0])
print(r[0,:,:,0])

import keras2onnx
import onnx

onnx_model = keras2onnx.convert_keras(b, target_opset=12)
onnx.save_model(onnx_model, "model.onnx")

import onnxruntime as rt

session = rt.InferenceSession("model.onnx")
```

### Expected behavior
I expected that onnx could convert the tensorflow max_pool_with_argmax operation into the onnx MaxPool operation.

**Comment by mpaillassa** [comment]
Hello, thank you for your answer. Indeed it is not supported in the latest keras-onnx release 1.7.0 but it has been added after. So installing from source makes it work!

**Comment by jcwchen** [comment]
Glad it works. Close this issue then. Please reopen it if you have other ONNX related questions. Thanks

**Comment by mpaillassa** [comment]
Hello, I think there is a bug with the tensorflow max_pool_with_argmax op.
When running the op with onnxruntime, I find that the op returns the pooled values twice instead of the pooled values and the pooled indices. Here is a code reproducing the bug:

`import tensorflow as tf
import numpy as np
  
class Bug(tf.keras.Model):   
    def __init__(self):`
        super(Bug, self).__init__()
        
    def call(self, inputs):
        v, i = tf.nn.max_pool_with_argmax(inputs, ksize=[1, 2, 2, 1], strides=[1, 2, 2, 1], padding='SAME')
        return tf.cast(i, tf.float32)

b = Bug()
np.random.seed(0)
inp = np.random.uniform(0, 10, (2, 4, 4, 3))
keras_indices = b(inp)

import onnx
import keras2onnx

onnx_model = keras2onnx.convert_keras(b, target_opset=12)
onnx.save_model(onnx_model, "test.onnx")

import onnxruntime as rt

session = rt.InferenceSession("test.onnx")
input, output = session.get_inputs()[0], session.get_outputs()[0]
onnx_indices = session.run([output.name], {input.name: inp.astype(np.float32)})[0]

print(inp[0,:,:,2])
print(onnx_indices[0,:,:,2])`

In the end it produces:
`[[6.02763376 6.45894113 9.63662761 5.2889492 ]
 [0.71036058 8.32619846 9.78618342 7.80529176]
 [1.43353287 4.1466194  4.56150332 6.17635497]
 [9.43748079 4.37031954 6.66766715 1.28926298]]
[[8.326199  9.786183 ]
 [9.437481  6.6676674]]`

Note that I had to cast the resulting indices as float otherwise I get the following error during inference:
`FAIL : Load model from test.onnx failed:Type Error: Type (tensor(int64)) of output arg (Identity:0) of node (bug/MaxPoolWithArgmax_transpose_2_1) does not match expected type (tensor(float)).`

I am not sure if it is an onnx or a converter issue.

**Comment by mpaillassa** [comment]
Should I rather open another issue about this ? @jcwchen

**Comment by jcwchen** [comment]
Yes. It seems to me that something wrong during the conversion. Please raise an issue in keras-onnx. Thanks.
